File: fixtures/native-stdlib.kt
fun lucentInternal_a2e2c7dbb96a1210_abs(value: Double): Double {
  return kotlin.math.abs(value)
}

fun lucentInternal_a2e2c7dbb96a1210_sqrt(value: Double): Double {
  return kotlin.math.sqrt(value)
}

fun lucentInternal_a2e2c7dbb96a1210_floor(value: Double): Double {
  return kotlin.math.floor(value)
}

fun lucentInternal_a2e2c7dbb96a1210_ceil(value: Double): Double {
  return kotlin.math.ceil(value)
}

fun lucentInternal_a2e2c7dbb96a1210_sin(value: Double): Double {
  return kotlin.math.sin(value)
}

fun lucentInternal_a2e2c7dbb96a1210_cos(value: Double): Double {
  return kotlin.math.cos(value)
}

fun lucentInternal_a2e2c7dbb96a1210_min(a: Double, b: Double): Double {
  return kotlin.math.min(a, b)
}

fun lucentInternal_a2e2c7dbb96a1210_max(a: Double, b: Double): Double {
  return kotlin.math.max(a, b)
}

fun lucentInternal_e08d398e59797131_trim(value: String): String {
  return value.trim()
}

fun lucentInternal_e08d398e59797131_contains(value: String, search: String): Boolean {
  return value.contains(search)
}

fun lucentInternal_d04f744ec3bed451_encodeUTF8(text: String): ArrayBuffer {
  return LucentBytes.fromByteArray(text.toByteArray(Charsets.UTF_8))
}

fun lucentInternal_d04f744ec3bed451_decodeUTF8(bytes: ArrayBuffer): String {
  return LucentBytes.toByteArray(bytes).toString(Charsets.UTF_8)
}

fun lucentInternal_d04f744ec3bed451_copyBytes(bytes: ArrayBuffer): ArrayBuffer {
  return LucentBytes.fromByteArray(LucentBytes.toByteArray(bytes))
}

fun magnitude(value: Double): Double {
  return lucentInternal_a2e2c7dbb96a1210_sqrt(lucentInternal_a2e2c7dbb96a1210_abs(value))
}

fun matches(value: String): Boolean {
  return lucentInternal_e08d398e59797131_contains(lucentInternal_e08d398e59797131_trim(value), "lucent")
}

fun roundTrip(text: String): String {
  return lucentInternal_d04f744ec3bed451_decodeUTF8(lucentInternal_d04f744ec3bed451_copyBytes(lucentInternal_d04f744ec3bed451_encodeUTF8(text)))
}

File: fixtures/native-stdlib.swift
import Foundation

func lucentInternal_736b35a22321c26_abs(value: Double) throws -> Double {
  return Swift.abs(value)
}

func lucentInternal_736b35a22321c26_sqrt(value: Double) throws -> Double {
  return value.squareRoot()
}

func lucentInternal_736b35a22321c26_floor(value: Double) throws -> Double {
  return value.rounded(.down)
}

func lucentInternal_736b35a22321c26_ceil(value: Double) throws -> Double {
  return value.rounded(.up)
}

func lucentInternal_736b35a22321c26_sin(value: Double) throws -> Double {
  return Foundation.sin(value)
}

func lucentInternal_736b35a22321c26_cos(value: Double) throws -> Double {
  return Foundation.cos(value)
}

func lucentInternal_736b35a22321c26_min(a: Double, b: Double) throws -> Double {
  return Swift.min(a, b)
}

func lucentInternal_736b35a22321c26_max(a: Double, b: Double) throws -> Double {
  return Swift.max(a, b)
}

func lucentInternal_2aaae6189ce4e2af_trim(value: String) throws -> String {
  return value.trimmingCharacters(in: .whitespacesAndNewlines)
}

func lucentInternal_2aaae6189ce4e2af_contains(value: String, search: String) throws -> Bool {
  return value.contains(search)
}

func magnitude(value: Double) throws -> Double {
  return try lucentInternal_736b35a22321c26_sqrt(value: lucentInternal_736b35a22321c26_abs(value: value))
}

func matches(value: String) throws -> Bool {
  return try lucentInternal_2aaae6189ce4e2af_contains(value: lucentInternal_2aaae6189ce4e2af_trim(value: value), search: "lucent")
}
